using_restricted_array.rb

```diff
@@ -8,28 +8,61 @@
 
 ## Calculates the length of the restricted integer array_size
 def length(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  while array[i]
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  while array[i]
+    puts array[i]
+    i += 1
+  end
 end
 
 # Reverses the values in the integer array
-def reverse(array, length) # Ruby
-  puts "NOT IMPLEMENTED"
+def reverse(array, length) #  Ruby
+  start = 0
+  last = (length(array) - 1)
+  while start <= last
+    array[start], array[last] = array[last], array[start]
+    start += 1
+    last -= 1
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  found = false
+  length.times do |i|
+    if value_to_find == array[i]
+      found = true
+      return found
+    end
+  end
+  return found
 end
 
 # Sorts the array in ascending order.
 def sort(array, length)
-  puts "NOT IMPLEMENTED"
+  #TODO

```

Comment:
See https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Sorting.md for more learning on this.